src/components/now-playing-drawer/Scrubber.tsx

```diff
@@ -0,0 +1,284 @@
+
+import moment from 'moment'
+import {
+  StyleSheet,
+  View,
+  GestureResponderEvent,
+  Animated,
+  PanResponder
+} from 'react-native'
+import LinearGradient from 'react-native-linear-gradient'
+
+import Text from 'app/components/text'
+import { usePressScaleAnimation } from 'app/hooks/usePressScaleAnimation'
+import { useThemedStyles } from 'app/hooks/useThemedStyles'
+import { ThemeColors, useThemeColors } from 'app/utils/theme'
+
+// How much the handle "grows" when pressing
+const HANDLE_GROW_SCALE = 1.1
+
+const SECONDS_PER_MINUTE = 60
+const MINUTES_PER_HOUR = 60
+const SECONDS_PER_HOUR = SECONDS_PER_MINUTE * MINUTES_PER_HOUR
+
+// Pretty formats seconds into m:ss or h:mm:ss
+const formatSeconds = (seconds: number) => {
+  const utc = moment.utc(moment.duration(seconds, 'seconds').asMilliseconds())
+  if (seconds > SECONDS_PER_HOUR) {
+    return utc.format('h:mm:ss')
+  }
+  return utc.format('m:ss')
+}
+
+const createStyles = (themeColors: ThemeColors) =>
+  StyleSheet.create({
+    root: {
+      flexDirection: 'row',
+      alignItems: 'center'
+    },
+    timestamp: {
+      color: themeColors.neutral,
+      fontSize: 12,
+      flexShrink: 1
+    },
+    slider: {
+      height: 4,
+      marginLeft: 16,
+      marginRight: 16,
+      flexGrow: 1
+    },
+    rail: {
+      backgroundColor: themeColors.neutralLight8,
+      borderRadius: 2,
+      borderColor: themeColors.neutralLight8,
+      height: 4,
+      overflow: 'hidden'
+    },
+    tracker: {
+      // The tracker must be the full width of the rail
+      // so that it can have rounded edges. It animates
+      // by sliding (translateX) instead of by growing.
+      flexGrow: 1,
+      backgroundColor: themeColors.neutral,
+      borderRadius: 2,
+      borderColor: themeColors.neutralLight8,
+      height: 4
+    },
+    handleContainer: {
+      top: -6,
+      position: 'absolute'
+    },
+    handle: {
+      marginLeft: -8,
+      height: 16,
+      width: 16,
+      borderRadius: 8,
+      backgroundColor: themeColors.staticWhite,
+      // Note: React-native-shadow-2 seems to lose fidelity
+      // when styling such a small shadow.
+      // TODO: Revisit this, but as of writing, these values
+      // are fairly good on android/ios
+      shadowColor: 'rgb(133,129,153)',
+      shadowOffset: {
+        width: 0,
+        height: 1
+      },
+      shadowOpacity: 0.5,
+      shadowRadius: 2,
+      elevation: 3
+    }
+  })
+
+const attachToDx = (animation: Animated.Value, newValue: number) => (
+  e: GestureResponderEvent
+) => {
+  Animated.event(
+    [
+      null,
+      {
+        dy: animation
+      }
+    ],
+    { useNativeDriver: false }
+  )(e, { dy: newValue })
+}
+
+type ScrubberProps = {
+  mediaKey: string
+}
+
+// Memoize because the Scrubber is sensitive to parent rerenders
+// due to its gesture handler use.
+// useSelectorWeb (used to connect the now playing drawer) induces
```

Comment:
this comment is great! do we have a pattern for block comment syntax btw?

Reply:
yeah we do. i should switch this 👍  ty